### PR TITLE
Identity-X GET and DELETE endpoints

### PR DESCRIPTION
**Pull request by Shinsina**
**REQUIRES: https://github.com/parameter1/base-cms/pull/878**

Screenshot from 2024-02-20 15-42-27
Screenshot from 2024-02-20 15-39-35
Screenshot from 2024-02-20 15-40-07
Screenshot from 2024-02-20 15-40-28
Screenshot from 2024-02-20 15-40-44
Screenshot from 2024-02-20 15-41-59
Screenshot from 2024-02-20 15-42-09

### CODE REVIEW

File: packages/braze/ingest/index.js [hunk 1]
```diff
@@ -124,4 +124,74 @@ module.exports = (app) => {
...
+        });
+      }
+      const userWasDeleted = await identityX.deleteAppUserForCurrentApplication({
+        ...(email && { email }),
+        ...(userId && { userId }),
+      });
+      if (userWasDeleted) {
+        res.json({
+          message: 'User was deleted for provided input',
+        });
```
Comment (solocommand) [inline packages/braze/ingest/index.js:184]: Return after send

File: packages/braze/ingest/index.js [hunk 2]
```diff
@@ -124,4 +124,74 @@ module.exports = (app) => {
...
+      const { email, userId } = req.body;
+      if (email && userId) throw Error('email XOR (exclusive or) userId is permitted!');
+      if (!email && !userId) throw Error('email XOR (exclusive or) userId is required!');
+      const userFromEmail = email ? await identityX.loadAppUserByEmail(email) : null;
+      const userById = userId ? await identityX.findUserById(userId) : null;
+      if (!userFromEmail && !userById) {
+        res.status(404).json({
+          user: {},
+          message: 'User not found for input',
+        });
```
Comment (solocommand) [inline packages/braze/ingest/index.js:175]: Return after sending the response

File: packages/braze/ingest/index.js [hunk 3]
```diff
@@ -124,4 +124,74 @@ module.exports = (app) => {
...
+      const { email, userId } = req.query;
+      if (email && userId) throw Error('email XOR (exclusive or) userId is permitted!');
+      if (!email && !userId) throw Error('email XOR (exclusive or) userId is required!');
+      const userFromEmail = email ? await identityX.loadAppUserByEmail(email) : null;
+      const userById = userId ? await identityX.findUserById(userId) : null;
+      if (userFromEmail) {
+        res.json({
+          user: userFromEmail,
+          message: `User found for email: ${email}`,
+        });
```
Comment (solocommand) [inline packages/braze/ingest/index.js:141]: return after send

**Review by solocommand — CHANGES_REQUESTED** [review]
Make sure you return (or otherwise skip) remaining code after sending a response, otherwise the following statements will still be executed, leading to `Cannot send headers...` errors server-side.

Otherwise, looks good! 👍🏾

**Review by B77Mills — APPROVED** [review]

**Pull request merged**